deploy/terraform/modules/app_tier/nsg-app-subnet.tf

```
@@ -0,0 +1,21 @@
+# Creates SAP app subnet nsg
+resource "azurerm_network_security_group" "nsg-app" {
+  count               = local.enable_deployment ? (var.infrastructure.vnets.sap.subnet_app.nsg.is_existing ? 0 : 1) : 0
+  name                = var.infrastructure.vnets.sap.subnet_app.nsg.name
+  location            = var.infrastructure.region
+  resource_group_name = var.vnet-sap[0].resource_group_name
```

Maybe I missed this in the previous review.
NSG should be deployed in the rg that is defined in the input but not tight to the vnet :)
```suggestion
  location                      = var.resource-group[0].location
  resource_group_name           = var.resource-group[0].name
```

---

Good catch, I just moved the code to the appropriate file.